src/main/scala/com/redhat/et/silex/util/optionalArg.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * This file is part of the "silex" library of helpers for Apache Spark.
+ *
+ * Copyright (c) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.c
+ */
+
+package com.redhat.et.silex.util
+
+/**
+ * Provides idiomatic optional parameter values on top of [[Option]], but which can be passed in as
+ * raw values, instead of having to pass in Some(value)
+ * {{{
+ * import com.redhat.et.silex.util.OptionalArg
+ *
+ * def possiblyFilter(data: Seq[Int], filterMax: OptionalArg[Int] = None) = {
+ *   // Treat an OptionalArg like an Option, for idiomatic handling of unset vals as None
+ *   if (filterMax.isEmpty) data else data.filter(_ <= filterMax.get)
+ * }
+ *
+ * val data = Seq(1, -1, 2)
+ *
+ * // default filterMax is None
+ * possiblyFilter(data) // returns data unfiltered
+ *
+ * // provide argument as (-1), instead of having to specify Some(-1)
+ * possiblyFilter(data, filterMax = -1) // returns Seq(-1)
+ *
+ * // Support full Option methods and implicit conversion to Option
+ * import OptionalArg.fullOptionSupport
+ * OptionalArg("xxx").map(_ + "zzz").get // returns "xxxzzz"
+ * }}}
+ *
+ * @tparam A The underlying value type.
+ * @param option The raw Option value.  May be implicitly constructed from raw value of type A
+ */
+class OptionalArg[+A](val option: Option[A]) extends AnyVal {
+  /** Determine if the optional argument is empty (contains None)
+   * @return true if the option is None, false otherwise
+   */
+  def isEmpty = option.isEmpty
+
+  /** Determine if the optional argument is defined (contains Some(value))
+   * @return true if option contains Some(value), false otherwise
+   */
+  def isDefined = option.isDefined
+
+  /** Get the value of the optional argument, assuming the argument is not empty.
+   * @return the value of the optional argument, or throw an exception if it is None
+   */
+  def get = option.get
+
+  /** Get the value of the optional argument if it is defined, otherwise return the given default.
+   * @param d A block evaluating to a default value, if the the optional value is not defined
+   * @return The value of the optional argument if one was defined, otherwise the value of d
+   */
+  def getOrElse[A1 >: A](d: => A1): A1 = option.getOrElse(d)
```

Comment:
In this case, it was just because that is how Option's type signature looks.  Not exactly sure why.  I saw something similar with signature of `unioned` for HashMap.